docs/preview/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -140,111 +131,105 @@ public class Startup
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
 
-        // Specify a topic subscription prefix instead of a name to separate topic message pumps.
-        services.AddServiceBusTopicMessagePumpWithPrefix(
-            "My-Service-Bus-Topic-Name"
-            "My-Service-Bus-Subscription-Prefix",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
 
-        // Uses managed identity to authenticate with the Service Bus Topic:
-        services.AddServiceBusTopicMessagePumpUsingManagedIdentity(
-            topicName: properties.EntityPath,
-            subscriptionName: "Receive-All", 
-            fullyQualifiedNamespace: "<your-namespace>.servicebus.windows.net"
-            // The optional client id to authenticate for a user assigned managed identity. More information on user assigned managed identities cam be found here:
-            // https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-a-user-assigned-managed-identity-works-with-an-azure-vm
-            clientId: "<your-client-id>");
+### Bring your own deserialization
+You can also choose to extend the built-in message deserialization with additional deserializer to meet your needs. 
+This allows you to easily deserialize into different message formats or reuse existing (de)serialization capabilities that you already have without altering the message router. 
 
-        services.AddServiceBusTopicMessagePump(
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING",
-            options => 
-            {
-                // Indicate whether or not messages should be automatically marked as completed 
-                // if no exceptions occurred and processing has finished (default: true).
-                options.AutoComplete = true;
+You start by implementing an `IMessageBodySerializer`. The following example shows how an expected type can be transformed to something else. 
+The result type (in this case `OrderBatch`) will be then be used to check if there is an `IAzureServiceBusMessageHandler` registered with that message type.
 
-                // Indicate whether or not the message pump should emit security events (default: false).
-                options.EmitSecurityEvents = true;
+```csharp
+using Arcus.Messaging.Pumps.Abstractions.MessageHandling;
 
-                // The amount of concurrent calls to process messages 
-                // (default: null, leading to the defaults of the Azure Service Bus SDK message handler options).
-                options.MaxConcurrentCalls = 5;
+public class OrderBatchMessageBodySerializer : IMessageBodySerializer
+{
+    public async Task<MessageResult> DeserializeMessageAsync(string messageBody)
```

Comment:
Should the blank line between 154 & 155 be removed ?

Reply:
Not sure what you're referring to. The line here doesn't contain a blank and the code sample doesn't have a blank besides between the usings and the class.

Reply:
Yeah, probably. I don't see this, but I got my browser zoom always on 75% 😅 . Formatting is best looked at in the deploy preview instead, since there's the final result.